.github/workflows/yocto-build-deploy.yml

```diff
@@ -293,19 +294,30 @@ jobs:
       #     regex: '^v20[0-9][0-9].[0-1]?[1470].[0-9]+$'
 
       # This will control the deployment of the hostapp only - it will determine if it is marked as final or not
-      # The hostapp being finalised is what determines if the API will present this OS version to users
+      # The hostapp being finalised is what determines if the API will present this OS version to user
+      # If the test_matrix is empty - it means there are no tests for the DT - so don't check tests, and don't finalise
+      # This check will check for a workflow with the same name as the yocto machine - so ensure that workflows are named after the yocto-machine
       - name: Check test results
-        if: ${{ inputs.check-merge-tests }} # Left in the case of manual deploys where tests are failing but we had to force merge
+        if: ${{ inputs.check-merge-tests && inputs.test_matrix != '{}' }} # Left in the case of manual deploys where tests are failing but we had to force merge
         id: merge-test-result
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
```suggestion
          # environment variables used by gh CLI
          # https://cli.github.com/manual/gh_help_environment
          GH_DEBUG: "true"
          GH_PAGER: "cat"
          GH_PROMPT_DISABLED: "true"
          GH_REPO: "${{ github.repository }}"
          GH_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
```

Reply:
These variables provide a cleaner experience using GH CLI in non-interactive scripts.